modules/utils/src/filesystem/file.cpp

```diff
@@ -5,20 +5,23 @@
 #include <fstream>
 #include <ios>
 #include <iterator>
+#include <optional>
 #include <span>
 #include <string>
 #include <string_view>
 #include <vector>
 
-#include <fmt/core.h>
-
-#include "hephaestus/utils/exception.h"
+#ifdef __linux__
+#include <linux/limits.h>
+#endif
+#include <unistd.h>
 
 namespace heph::utils::filesystem {
-auto readFile(const std::filesystem::path& path) -> std::string {
+auto readFile(const std::filesystem::path& path) -> std::optional<std::string> {
   std::ifstream infile{ path };
-  throwExceptionIf<InvalidDataException>(!infile,
-                                         fmt::format("Could not open file {} to read", path.string()));
+  if (!infile) {
+    return {};
```

Comment:
I guess we should add logging here, e.g. `LOG(WARNING)`...

Reply:
if we had c++23 we could return `std::expect` with the error message, but I guess for now we need to settle on the warning.